src/services/axios-http-data.service.ts

```diff
@@ -75,53 +84,67 @@ export class AxiosHttpDataService extends AbstractHttpDataService {
    * @param url The URL of the endpoint to call
    * @param queryParams Any query Params to send
    * @param headers any HTTP Headers to send
+   * @param timeout The number of milliseconds that an API request should wait before timing out.
+   * @param retries The number of times to retry the operation, if a timeout is received.
    * @param axiosRequestCallFn The axios operation function
    */
   private async axiosHttpCall<K>(
     url: string,
     queryParams: IQueryParams,
     headers: IHeaders,
+    timeout: number,
+    retries: number,
     axiosRequestCallFn: (fnUrl: string, fnRequestConfig: AxiosRequestConfig) => Promise<AxiosResponse<K>>
   ): Promise<IApiResponse<K>> {
 
     const requestConfig: AxiosRequestConfig = {
       headers,
-      params: queryParams
+      params: queryParams,
+      timeout
     };
 
-    try {
+    for (let attempt = 0; attempt <= retries; attempt++) {
```

Comment:
As discussed, I feel there is a better way to implement these retry handlers without attempting to natively integrate code into the core functionality. Is there a plan to implement the expected retry pattern?

Reply:
Discussed offline